opensrp-giz-malawi/src/main/java/org/smartregister/giz/configuration/OpdRegisterQueryProvider.java

```diff
@@ -47,13 +53,14 @@ public String getObjectIdsQuery(@Nullable String filters, @Nullable String mainC
             }
         } else if (!TextUtils.isEmpty(mainCondition)) {
             String sqlQuery =
-                    "Select ec_client.id as object_id, ec_client.last_interacted_with, (opd_details.current_visit_start_date IS NOT NULL AND opd_details.current_visit_start_date >= '$latest_start_visit_date' AND opd_details.current_visit_end_date IS NULL) AS checked_in FROM ec_client \n" +
-                    "    LEFT JOIN opd_details ON ec_client.base_entity_id = opd_details.base_entity_id "+
-                    "inner join client_register_type crt on crt.base_entity_id = ec_client_search.object_id  " +
-                    " WHERE crt.register_type = 'opd' AND checked_in \n" +
-                    "ORDER BY last_interacted_with DESC";
+                    "Select ec_client.id as object_id\n" +
+                            "FROM ec_client\n" +
+                            "inner join client_register_type crt on crt.base_entity_id = ec_client.id  \n" +
+                            "WHERE crt.register_type = 'opd'\n" +
+                            "and  ec_client.id IN (SELECT base_entity_id FROM opd_client_visits where visit_group = '" + todayDate + "')\n" +
+                            "ORDER BY last_interacted_with DESC";
```

Comment:
You could use a join here instead of a subquery.

Reply:
Refactored